packages/editor/src/lib/editor/managers/LicenseManager.ts

```diff
@@ -50,16 +51,43 @@ export interface ValidLicenseKeyResult {
 
 type TestEnvironment = 'development' | 'production'
 
+/** @internal */
 export class LicenseManager {
 	private publicKey =
 		'MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEHJh0uUfxHtCGyerXmmatE368Hd9rI6LH9oPDQihnaCryRFWEVeOvf9U/SPbyxX74LFyJs5tYeAHq5Nc0Ax25LQ=='
 	public isDevelopment: boolean
 	public isCryptoAvailable: boolean
+	state = atom('license state', 'pending')
 
-	constructor(testPublicKey?: string, testEnvironment?: TestEnvironment) {
+	constructor(
+		licenseKey: string | undefined,
+		testPublicKey?: string,
+		testEnvironment?: TestEnvironment
+	) {
 		this.isDevelopment = this.getIsDevelopment(testEnvironment)
 		this.publicKey = testPublicKey || this.publicKey
 		this.isCryptoAvailable = !!crypto.subtle
+
+		this.getLicenseFromKey(licenseKey).then((result) => {
+			function isEditorUnlicensed() {
```

Comment:
we need to make sure this stuff is tested, it's pretty key

Reply:
Extracted this to a separate function. I'll adapt the tests to make sure it produces the right output.